src/pycountry/__init__.py

```diff
@@ -170,7 +170,10 @@ def __init__(self, **kw):
         super().__init__(**kw)
         self.country_code = self.code.split("-")[0]
         if self.parent_code is not None:
-            self.parent_code = f"{self.country_code}-{self.parent_code}"
+            # Split the parent_code to check if the country_code is already present
```

Comment:
Is this a data quality issue in upstream? Then we should report it back with the Debian maintainers.

Reply:
@ctheune I guess I'm not sure if it is a data quality issue or not. Here is some data from in `ISO3166-2.json`

``` json
    {
      "code": "GB-BAS",
      "name": "Bath and North East Somerset",
      "parent": "GB-ENG",
      "type": "Unitary authority"
    },
    {
      "code": "GB-ENG",
      "name": "England",
      "type": "Country"
    },
    {
      "code": "US-AL",
      "name": "Alabama",
      "type": "State"
    }
```

Data for Great Britain is odd because you have a very deep hierarchical structure where a Unitary Authority rolls up to a Country that isn't part of `ISO 3166-1` due to the recognized `ISO 3166-1` country being the `United Kingdom`.

So this SEEMS right to me, but you might know better. 

The issue is that when this code was ran for `GB-BAS` it would return `GB`.

``` python
        self.country_code = self.code.split("-")[0]
```

It would then proceed to the following code, which when run would pass the if check and then append the `self.country_code` = `GB` to the self.parent_code `GB-ENG` and add a hyphen between resulting in the `GB-GB-ENG`

``` python
        if self.parent_code is not None:
            self.parent_code = f"{self.country_code}-{self.parent_code}"
```

By adding in this code, instead of evaluating that `self.parent_code` is not none it now ensures that `parts[0]` (new `self.parent_code` alias), in this case `GB` is not equal to the `self.country_code`, also `GB`

```
    parts = self.parent_code.split("-")
            if parts[0] != self.country_code:
                self.parent_code = f"{self.country_code}-{self.parent_code}"
```

To go back to our `GB-BAS` example and our `US-AL` example it would then return parent_codes of `GB-ENG` and `None` as is expected. 

I was wondering if the appending of the country_code to the front of the parent_code was necessary but I have confirmed that it is. Subdivisions such as `BD-36` has a `parent` of `C`, not including the `BD-` prefix. So it is necessary in this case to still append the `country_code` `BD-` to the front of `C`. Which you can see works correctly in production

``` python
pycountry.subdivisions.get(code="BD-36")
SubdivisionHierarchy(code='BD-36', country_code='BD', name='Madaripur', parent='C', parent_code='BD-C', type='District')
```

Reply:
@ctheune let me know if you've had a chance to look at this if so will likely merge.